[18.0][MIG] account_financial_report: Migration to 18.0

chaule97 opened the pull request:
- OCA Port: https://github.com/OCA/account-financial-reporting/pull/1233
- I have renamed [account name](https://github.com/chaule97/account-financial-reporting/blob/befa9ceabbc8379c92dd03f4cf206ebcc11604c1/account_financial_report/tests/test_trial_balance.py#L77) "301" to "201" in `TestTrialBalanceReport` based on [SQL logic](https://github.com/odoo/odoo/blob/b20d26ca523a9971c05789256901b10bf2b488de/addons/account/models/account_account.py#L456) matching account codes to group prefixes.
- I have updated the test case test_05_all_accounts_loaded to only retrieve accounts with numeric codes, because there is [account](https://github.com/odoo/odoo/blob/b20d26ca523a9971c05789256901b10bf2b488de/addons/account/tests/common.py#L167) with string codes.

pedrobaeza committed:
[FIX] account_financial_report: Don't fail with installed l10n

The number of computed account can be greater in certain databases with l10n modules.

pedroslopez committed:
[FIX] account_financial_report: Update to bootstrap 4

eduaparicio committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (329 of 329 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_financial_report/pt_BR/

Commit:
Translated using Weblate (French)

Currently translated at 10.3% (34 of 329 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_financial_report/fr/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (329 of 329 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_financial_report/pt_BR/

pedrobaeza committed:
[FIX] account_financial_report: Avoid slow vacuum due to ondelete=cascade

`report_journal_ledger` is auto-vacuumed as any transient model, but has some
ondelete="cascade" constraints that auto-remove subtables when a record is removed,
doing this operation very slow when selecting these sub-records.

Letting default ondelete="set null" would result in same performance bottleneck,
as the select on sub-table is performed the same for setting "null" value on them.

As a solution, and for avoiding a costly index operation, we delete by SQL sub-tables
rows in advance.

A bit of extra logic has been added for avoiding to remove that records if it's not
the turn of vacuum the parent table.

Commit:
Fix display of captions in pdf

gaikaz committed:
account_financial_report: fix Trial Balance line order

Commit:
Translated using Weblate (French)

Currently translated at 23.1% (76 of 329 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_financial_report/fr/

Commit:
Translated using Weblate (French)

Currently translated at 23.1% (76 of 329 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_financial_report/fr/

sbejaoui committed:
[12.0][FIX] - Hide ending balance if general ledger is filtered by partner

Martronic-SA committed:
Translated using Weblate (French)

Currently translated at 84.5% (278 of 329 strings)

Translation: account-financial-reporting-12.0/account-financial-reporting-12.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-12-0/account-financial-reporting-12-0-account_financial_report/fr/

JoanSForgeFlow committed:
[12.0] account_financial_report: refactor

badbole committed:
Added translation using Weblate (Croatian)

grindtildeath committed:
account_financial_report: Increase columns width in general ledger
Before this commit, amounts such as "10'000.00 CHF" were exceeding their
columns width. Label column could be reduced as it's printed correctly
over two different lines.

Commit:
[IMP] account_financial_report: black, isort

Commit:
[MIG] account_financial_report: Migration to 13.0

JoanSForgeFlow committed:
[FIX] account_financial_report: general_ledger
* centralization is only calculated on accounts where 'centralized' is True
* Cumul. Balance of each move_line fixed
* Ordering move_lines by date
* Fix trial_balance xlsx when not show_partner_details

Commit:
[FIX] account_financial_report: Fix error when generating reports

Some reports crash if account move line implicated are Sections lines or Notes lines in an invoice

traviswaelbro committed:
[FIX] account_financial_report: open_items

* Missing import causes failure when running Open Items with Date At in the past

JoanSForgeFlow committed:
[IMP] account_financial_report: open_items
* added option "show_partner_details"

JoanSForgeFlow committed:
[IMP] account_financial_report: open_items & aged_partner_balance
* Now account_ids is empty by default and required
* Domain is applied in the selection of the accounts

JoanSForgeFlow committed:
[IMP] account_financial_report: select accounts between two codes

JordiBForgeFlow committed:
[IMP] account_financial_report: multiple fixes and improvements

Tonow-c2c committed:
fixup! account_financial_report: Increase columns width in general ledger Before this commit, amounts such as "10'000.00 CHF" were exceeding their columns width. Label column could be reduced as it's printed correctly over two different lines.

JoanSForgeFlow committed:
[IMP] account_financial_report: Fix VAT Report and other improvements

JoanSForgeFlow committed:
[IMP] account_financial_report:
* filter by analytic_account

JordiBForgeFlow committed:
[IMP] account_financial_report:
* fix speed issue in VAT report
* rename Cost Center to Analytic Account in General Ledger

JoanSForgeFlow committed:
[FIX] account_financial_report:
Not showing journal items with amount residual equal to 0

JoanSForgeFlow committed:
[IMP] account_financial_report: general_ledger
* "hide account at 0" hides account/partner if inital cumul balance is 0 and there are not movements in the period

Commit:
FIX missing date in invoice

The second invoice created has no explicit so taking the current date could lead to failing tests because the invoice is out of search as the the setup sets date_from as time.strftime("%Y-%m-01") and date_to as time.strftime("%Y-%m-28"). So i choose the easy way by setting an explicit date for the invoice.

astirpe committed:
[FIX] translation not loaded

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (229 of 229 strings)

Translation: account-financial-reporting-13.0/account-financial-reporting-13.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-13-0/account-financial-reporting-13-0-account_financial_report/es_AR/

sbidoul committed:
[FIX] duplicates in es.po

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (241 of 241 strings)

Translation: account-financial-reporting-13.0/account-financial-reporting-13.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-13-0/account-financial-reporting-13-0-account_financial_report/es_AR/

JoanSForgeFlow committed:
[ADD] domain field for additional filtering

JoanSForgeFlow committed:
[IMP] domain widget performance fix

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (242 of 242 strings)

Translation: account-financial-reporting-13.0/account-financial-reporting-13.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-13-0/account-financial-reporting-13-0-account_financial_report/es_AR/

pedrobaeza committed:
[FIX+IMP] account_financial_report: data can be None + no context mangling

Depending on the calling method, data can be something different from a dictionary
(like `None` if no keyword argument is provided), so this will crash. Example:
mis_builder is crashing in its tests.

We prevent that checking for a falsy value before trying to access the dictionary.

We also avoid context mangling for non AFR reports (those without the key
`account_financial_report_lang` in the dictionary), not callign to `with_context`
in that case.

JoanSForgeFlow committed:
[FIX] General ledger XLSX: solved mistake in the info

dhongu committed:
Translated using Weblate (Romanian)

Currently translated at 75.5% (185 of 245 strings)

Translation: account-financial-reporting-13.0/account-financial-reporting-13.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-13-0/account-financial-reporting-13-0-account_financial_report/ro/

dhongu committed:
Translated using Weblate (Romanian)

Currently translated at 78.0% (191 of 245 strings)

Translation: account-financial-reporting-13.0/account-financial-reporting-13.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-13-0/account-financial-reporting-13-0-account_financial_report/ro/

dhongu committed:
Translated using Weblate (Romanian)

Currently translated at 79.2% (194 of 245 strings)

Translation: account-financial-reporting-13.0/account-financial-reporting-13.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-13-0/account-financial-reporting-13-0-account_financial_report/ro/

MiquelRForgeFlow committed:
[UPD] Eficent -> ForgeFlow

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (245 of 245 strings)

Translation: account-financial-reporting-13.0/account-financial-reporting-13.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-13-0/account-financial-reporting-13-0-account_financial_report/es_AR/

tupaq committed:
[IMP] account_financial_report: black, isort, prettier

LoisRForgeFlow committed:
[13.0][FIX] account_financial_report: Correct all uses of
self.env.user.company_id. Also move company_id to the
wizard abstract model.

JoanSForgeFlow committed:
[FIX] account_financial_report: filter cancelled journal items

JoanSForgeFlow committed:
[IMP] account_financial_report: abstract for avoiding duplicated code and increase readability

StefanRijnhart committed:
[UPD] Update account_financial_report.pot

[a0cb11bb] [FIX] Journal ledger wizard to inherit from abstract report wizard

Fixes
    AttributeError: type object 'journal.ledger.report.wizard' has no attribute
    'button_export_xlsx'

Regression of #735

joao-p-marques committed:
[MIG] account_financial_report: Migration to 14.0

Since v14, Odoo defines the `__slots__` attribute in the `BaseModel` class (https://github.com/odoo/odoo/commit/ea3e39506a2b360545ec0cd1d3235e69ad16ea6c)
This makes it impossible to add attributes to an instance like it was done here in v13.
The use of the `report_data` dictionary passed between method is the closes and simples solution to this "issue".

TT26415

Co-authored-by: Alex Cuellar <acuellar@grupoyacck.com>

joao-p-marques committed:
[FIX] account_financial_report: Fix tests

On multi-company settings, we need to make sure we are searching and creating objects with the correct company associated.
Also take into account previously created account groups

 TT28423

joao-p-marques committed:
[FIX] account_financial_report: Refactor tests

Use core account test base, helping to ensure we have the correct accounting environment setup.
If any of the localization modules are loaded, this will make this tests be skipped, instead of simply failing

TT28423

anasuarez1 committed:
Translated using Weblate (Spanish)

Currently translated at 94.2% (231 of 245 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/es/

alan196 committed:
Added translation using Weblate (Spanish (Mexico))

alan196 committed:
Translated using Weblate (Spanish (Mexico))

Currently translated at 0.0% (0 of 245 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/es_MX/

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 93.4% (229 of 245 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/fr/

Tardo committed:
[FIX] account_financial_report: Actions

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (245 of 245 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (245 of 245 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/es_AR/

Commit:
[FIX] Default company should not be used since it doesn't mutate when switching companies.

Yvesldff committed:
Added translation using Weblate (French (France))

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 97.1% (238 of 245 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/fr_FR/

JordiMForgeFlow committed:
[FIX] account_financial_report: fix VAT report template

MiquelRForgeFlow committed:
[FIX] account_financial_report: fix test

Due to https://github.com/odoo/odoo/commit/fcaa54939e9a4f0dd5e47cd0ccffe7aa24bd451c

joao-p-marques committed:
[FIX] account_financial_report: allow navigation on all fields

Pass res_id correctly to lines to allow navigation on the fields

In the open items ledger, the res_id attribute of the invoice
in the report was getting: (id, move_name)
The result was that, when clicking the line, it would redirect
to a new record, instead of the existing.
This passes only the id to the line, solving that issue.

Fix menu item name

TT29371

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 75.5% (185 of 245 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/nl/

joao-p-marques committed:
[FIX] account_financial_report: Fix tests

Add date to invoice to be able to post

joao-p-marques committed:
[FIX] account_financial_report: Don't include unaffected account if accounts are filtered

When generated a General Ledger or a Trial Balance, if some accounts are already filtered in the wizard don't show the information related to the unaffected earnings account

TT29658

joao-p-marques committed:
[FIX] account_financial_report: filter by correct data in trial balance

When on a Trial Balance report and filtering by hierarchical accounts or for a foreing currency, an error was produced when generating the report.

The report fields were dependant on data that is no longer present, such as line data.

This uses the correct data provided when generating the report to avoid this errors

TT29261

Zar21 committed:
[14.0][FIX] account_financial_report: Incorrect format definition

JordiBForgeFlow committed:
[account_financial_report][fix] display correct target_move in Trial Balance XLSX

ValentinVinagre committed:
[IMP] account_financial_report: added 'auto sequence' option in Journal Ledger

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (248 of 248 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/es_AR/

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (248 of 248 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/pt/

ferran-S73 committed:
[14.0][FIX] account_financial_report: fix method parameter order

ferran-S73 committed:
[14.0][IMP] account_financial_report: add keyword to function calls

SimoRubi committed:
[FIX] account_financial_report: Print report with details

ernestotejeda committed:
Translated using Weblate (Spanish)

Currently translated at 91.5% (227 of 248 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/es/

Commit:
[14.0][FIX] account_financial_reporting: wrong view_id

joao-p-marques committed:
[FIX] account_financial_report: Filter column creation to have correct offset in Journal Ledger

Fixes https://github.com/OCA/account-financial-reporting/issues/815
Seems like a missing change from https://github.com/OCA/account-financial-reporting/commit/0e3b4d29722b9018b58d611b16af945279790265

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 66.5% (165 of 248 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/it/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (249 of 249 strings)

Translation: account-financial-reporting-14.0/account-financial-reporting-14.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-14-0/account-financial-reporting-14-0-account_financial_report/es_AR/

fredzamoabg committed:
[14.0][FIX] account_financial_report: sort based partner name in open_items report

i-vyshnevska committed:
[FIX] account_financial_report: use proper methods to set context

Commit:
[14.0][FIX] account_financial_report: Run compute in multi

Commit:
[FIX] account_financial_report: Trial Balance with 'hide account at 0'

JasminSForgeFlow committed:
[IMP] account_financial_report: black, isort, prettier

JasminSForgeFlow committed:
[MIG] account_financial_report: Migration to 15.0

omar7r committed:
[IMP] account_financial_report: Completes the functionality of hide_parent_hierarchy_level field and fix the level of accounts for trial balance

Commit:
[14.0][FIX] account_financial_report: Memory problems setting filters
Fixes #814

jabelchi committed:
Translated using Weblate (Catalan)

Currently translated at 85.1% (212 of 249 strings)

Translation: account-financial-reporting-15.0/account-financial-reporting-15.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-15-0/account-financial-reporting-15-0-account_financial_report/ca/ (+1 squashed commits)

Commit:
[FIX] account_financial_report: trial balance hierarchy

When the report is generated with 'hierarchy' option checked,
the group hierarchy values are not calculated correctly

Commit:
[IMP] account_financial_report: Remove 'Computed Accounts' option from 'trial balance'.

LoisRForgeFlow committed:
[IMP] account_financial_report: common format for all amounts

In the open items report, the currency original and residual
fields were being displayed as regular float fields. With
this change they are printed with the corresponding format
for the currency of the journal item.

LoisRForgeFlow committed:
[FIX] account_financial_report: run test after install

Due to the change at https://github.com/odoo/odoo/commit/15a1b3ae5cb1117e2c1bda5058d3cd99f00a6769
it is now required to run tests that require CoA after install.

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 91.5% (228 of 249 strings)

Translation: account-financial-reporting-15.0/account-financial-reporting-15.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-15-0/account-financial-reporting-15-0-account_financial_report/es/

victoralmau committed:
[IMP] account_financial_report: Remove "Cur." column (currency is now added in "Amount cur.") from general_ledger.
TT38722

victoralmau committed:
[IMP] account_financial_report: Add "Cumul cur." column to general_ledger.
TT38722

victoralmau committed:
[FIX] account_financial_report: Sort journal ledger by move_id

Without this sort, and using auto-sequence, the numbers of the journal
entries may appear unsorted on the journal ledger. Forcing at least this
minimum sorting, we assure the items are sorted.

TT38775

victoralmau committed:
[IMP] account_financial_report: Hide values that have the same currency as the company currency

An amount_different_company_currency cell type is added to apply the following logic:
- If the currency is the same as the company's currency, we will leave the value empty.
- If the currency is different from the company's currency, we will show the value.

TT38722

victoralmau committed:
[IMP] account_financial_report: Refactor

TT38721

victoralmau committed:
[IMP] account_financial_report: Add tax_line_id field to taxes column in general_ledger

TT38721

alexis-via committed:
[FIX] account_financial_report: partner trial balance: sort by partner name

victoralmau committed:
[IMP] account_financial_report: Add tax grouping in general ledger

TT38721

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (251 of 251 strings)

Translation: account-financial-reporting-15.0/account-financial-reporting-15.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-15-0/account-financial-reporting-15-0-account_financial_report/es_AR/

ibuioli committed:
Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (258 of 258 strings)

Translation: account-financial-reporting-15.0/account-financial-reporting-15.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-15-0/account-financial-reporting-15-0-account_financial_report/es_AR/

are-agilebg committed:
Fixed the amount_residual_currency on partial reconcile

victoralmau committed:
[FIX] account_financial_report: Don't show cancelled items when clicking on interactive cells

TT39954

victoralmau committed:
[FIX] account_financial_report: Set the correct value of column 'Cumul cur.' in general ledger

TT39754

victoralmau committed:
[IMP] account_financial_report: Improve domain to reduce code in general ledger.

TT39754

victoralmau committed:
[FIX] account_financial_report: Change partner_domain to misc_grouped_domain in general_ledger.

TT39754

AaronHForgeFlow committed:
[IMP] account_financial_report: extensibility: ml fields

AaronHForgeFlow committed:
[IMP] account_financial_report: extensibility xlsx: allow to use other column types

victoralmau committed:
[FIX] account_financial_report: Show records in general ledger when not grouped.

ramiadavid committed:
[MIG] account_financial_report: Migration to 16.0

ramiadavid committed:
[IMP] account_financial_report: change report access groups to only 'Show Full Accounting Features'

Commit:
[UPD] Update account_financial_report.pot

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/

Reyes4711-S73 committed:
[15.0][FIX] account_financial_report: generate ledger document

In general ledger report shows an error due to the fact that the ball_curr varible does not generated

Reyes4711-S73 committed:
[16.0][FIX] account_financial_report: generate ledger document

In general ledger report shows an error due to the fact that the ball_curr varible does not generated

OCA-git-bot committed:
account_financial_report 16.0.1.0.1

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 70.2% (177 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/hr/

victoralmau committed:
[FIX] account_financial_report: General ledger changes:
- Fix total_bal_curr value from General ledger to prevent wrong values
- Prevent wrong values

TT41041

victoralmau committed:
[IMP] account_financial_report: Trial balance changes:

- Show Initial balance and Ending balance only when it is necessary with currency + hide currency column
- Show currency in the correct column
- Create a dict with all keys when it does not exist
- Prevent wrong values

TT41158

OCA-git-bot committed:
account_financial_report 16.0.1.0.2

victoralmau committed:
[IMP] account_financial_report: Remove amount_different_company_currency type from xlsx

OCA-git-bot committed:
account_financial_report 16.0.1.1.0

Commit:
[UPD] Update account_financial_report.pot

OCA-git-bot committed:
account_financial_report 16.0.1.2.0

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/

yajo committed:
[FIX] account_financial_report: endless installation

Installing the module in a DB with more than 50000 accounts and groups made the install stall.

It turns out this method's implementation produced almost endless recursion. it was also depending on some fields that never were used.

Now it depends on the parent path, so when one group is moved to another parent, its computed accounts are recomputed, and parents' too, recursively.

Now, the method is much more performant, and the module gets installed in the same DB in 30s.

@moduon MT-1900

Co-authored-by: Eduardo De Miguel <edu@moduon.team>

OCA-git-bot committed:
account_financial_report 16.0.1.2.1

douglascstd committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 78.5% (198 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/pt_BR/

OCA-git-bot committed:
account_financial_report 16.0.1.2.2

tate11 committed:
Added translation using Weblate (Turkish)

tate11 committed:
Translated using Weblate (Turkish)

Currently translated at 77.3% (195 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/tr/

pedrobaeza committed:
[FIX] account_financial_report: Assure the aml order

If we don't assure the order for the search, there are chances that
several items for the same account and date appear in a incorrect order:

24/02/2023 - BNK1/2023/02/0011 - ...
27/02/2023 - BNK1/2023/02/0013 - ...
27/02/2023 - BNK1/2023/02/0012 - ...
29/02/2023 - BNK1/2023/02/0014 - ...

TT42098

OCA-git-bot committed:
account_financial_report 16.0.1.2.3

florian-dacosta committed:
[FIX] account_financial_report: Account group styling in qweb and xlsx

OCA-git-bot committed:
account_financial_report 16.0.1.3.0

RachidAlassir committed:
[FIX] account_financial_report: Replace wrong field `amount_currency` with two fields `debit_amount_currency` and `credit_amount_currency`

gerard-vacas committed:
Translated using Weblate (Spanish)

Currently translated at 98.0% (247 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/es/

OCA-git-bot committed:
account_financial_report 16.0.1.3.1

Commit:
[FIX] account_financial_report fix to take the id of the currency if is set, that fix the following error:  opt/odoo/odoo/odoo/models.py", line 3681, in write
    field.write(self, vals[fname])
  File "/opt/odoo/odoo/odoo/fields.py", line 3057, in write
    return self.write_batch([(records, value)])
  File "/opt/odoo/odoo/odoo/fields.py", line 3078, in write_batch
    return self.write_real(records_commands_list, create)
  File "/opt/odoo/odoo/odoo/fields.py", line 3254, in write_real
    flush()
  File "/opt/odoo/odoo/odoo/fields.py", line 3216, in flush
    comodel.create(to_create)
  File "<decorator-gen-69>", line 2, in create
  File "/opt/odoo/odoo/odoo/api.py", line 348, in _model_create_multi
    return create(self, arg)
  File "/opt/odoo/odoo/odoo/addons/base/models/ir_fields.py", line 534, in create
    recs = super().create(vals_list)
  File "<decorator-gen-13>", line 2, in create
  File "/opt/odoo/odoo/odoo/api.py", line 348, in _model_create_multi
    return create(self, arg)
  File "/opt/odoo/odoo/odoo/models.py", line 3886, in create
    records = self._create(data_list)
  File "/opt/odoo/odoo/odoo/models.py", line 3992, in _create
    cr.execute(query, params)
  File "<decorator-gen-3>", line 2, in execute
  File "/opt/odoo/odoo/odoo/sql_db.py", line 101, in check
    return f(self, *args, **kwargs)
  File "/opt/odoo/odoo/odoo/sql_db.py", line 301, in execute
    res = self._obj.execute(query, params)
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/odoo/odoo/odoo/http.py", line 641, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/odoo/odoo/http.py", line 317, in _handle_exception
    raise exception.with_traceback(None) from new_cause
psycopg2.ProgrammingError: can't adapt type 'res.currency'

OCA-git-bot committed:
account_financial_report 16.0.1.3.2

victoralmau committed:
[IMP] account_financial_report: Show move_name in entry column from general ledger report

It is not necessary to repeat part of the text shown in Ref - Label column.

TT42731

victoralmau committed:
[FIX] account_financial_report: Prevent error related to currency from General ledger

Use case: Generate report (showing foreign currency) with accounts with defined currency.

Traceback:

File "/odoo/odoo-server/odoo/addons/base/models/ir_qweb_fields.py", line 448, in value_to_html
fmt = "%.{0}f".format(display_currency.decimal_places)
AttributeError: 'int' object has no attribute 'decimal_places'

The above exception was the direct cause of the following exception:

Error to render compiling AST
AttributeError: 'int' object has no attribute 'decimal_places'
Template: account_financial_report.report_general_ledger_lines
Path: /t/div/div[2]/t[6]/t[1]/div[1]/t[1]/span/t
Node: <t t-raw="account_or_group_item_object['init_bal']['bal_curr']" t-options="{'widget': 'monetary', 'display_currency': account['currency_id']}"/>

TT42804

[FIX] account_financial_report: Prevent error related to currency from General ledger

Use case: Generate xlsx report (showing foreign currency) with accounts with defined currency.

Traceback:

File "/opt/odoo/auto/addons/account_financial_report/report/abstract_report_xlsx.py", line 535, in _get_currency_amt_format
    field_name = "{}_{}".format(field_prefix, currency.name)
AttributeError: 'int' object has no attribute 'name'

ramiadavid committed:
[FIX] account_financial_report: Replace deprecated t-raw to t-out

ramiadavid committed:
[FIX] account_financial_report: Fix report links with domain

OCA-git-bot committed:
account_financial_report 16.0.1.3.3

OCA-git-bot committed:
account_financial_report 16.0.1.3.4

mymage committed:
Translated using Weblate (Italian)

Currently translated at 62.6% (158 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/it/

sanrav committed:
Translated using Weblate (French (France) (fr_FR))

Currently translated at 100.0% (252 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/fr_FR/

sanrav committed:
Translated using Weblate (French)

Currently translated at 100.0% (252 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/fr/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 62.6% (158 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/it/

OCA-git-bot committed:
account_financial_report 16.0.1.3.5

Commit:
[FIX] account_financial_report: fixing bug when calculation of credit is dublicated

SirTakobi committed:
[FIX] account_financial_report: Empty format

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (252 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/es/

victoralmau committed:
[FIX] account_financial_report: Prevent error related to currency from Trial Balance

Use case: Generate report (showing foreign currency) with accounts with defined currency.

Traceback:

Error to render compiling AST
AttributeError: 'int' object has no attribute 'decimal_places'
Template: account_financial_report.report_trial_balance_line
Path: /t/div/t[3]/t[1]/t/t/div/span/t
Node: <t t-esc="balance['initial_currency_balance']" t-options="{'widget': 'monetary', 'display_currency': balance['currency_id']}"/>

The error occured while rendering the template account_financial_report.report_trial_balance_line and evaluating the following expression: <t t-esc="balance['initial_currency_balance']" t-options="{'widget': 'monetary', 'display_currency': balance['currency_id']}"/>

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 89.2% (225 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/pt_BR/

OCA-git-bot committed:
[UPD] README.rst

aritzolea committed:
[FIX] account_financial_report: Initial currency balance value on trial balance report

OCA-git-bot committed:
oca-github-bot post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/

victoralmau committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (252 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/es/

josep-tecnativa committed:
[IMP] account_financial_report: test performance improvement

OCA-git-bot committed:
[BOT] post-merge updates

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 89.6% (226 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/pt_BR/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 62.6% (158 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/it/

grindtildeath committed:
Avoid error on installation with missing analytic account

When installing the module account_financial_report, a relational table between
account_move_line and account_analytic_account is created and computed.

However, if an analytic account was used only on draft invoices before being deleted,
its ID will remain in the JSON column analytic_distibution of account_move_line.
In that case we get a ForeignKeyViolation because the ID doesn't exist in
account_analytic_account table.

Therefore, we need to check if the ID exists during the computation to avoid
inserting it in the relational table and raising the error.

OCA-git-bot committed:
[BOT] post-merge updates

jakobkrabbe committed:
Added translation using Weblate (Swedish)

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 14.2% (36 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/sv/

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 98.4% (248 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/pt_BR/

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (252 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/fr/

Commit:
Translated using Weblate (Dutch)

Currently translated at 84.9% (214 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/nl/

adrianojprado committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 98.8% (249 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/pt_BR/

OCA-git-bot committed:
[BOT] post-merge updates

pedrobaeza committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (252 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/es/

ChristophAbenthungCibex committed:
[16.0][FIX]account_financial_report: missing accounts

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 57.9% (146 of 252 strings)

Translation: account-financial-reporting-16.0/account-financial-reporting-16.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-16-0/account-financial-reporting-16-0-account_financial_report/sv/

norlinhenrik committed:
[IMP] account_financial_report: pre-commit auto fixes

norlinhenrik committed:
[MIG] account_financial_report: Migration to 17.0

yajo committed:
[FIX] account_financial_report: optimize computation of analytic accounts

Installing the module in a big DB was very slow.

I applied several optimizations to the method for working better with big datasets:
- Prefetch all analytic account distribution.
- Batch writes as much as possible.

@moduon MT-4982

victoralmau committed:
[IMP] account_financial_report: Add Missing partner support in trial balance

If there are records without partner_id set and we show the detail of partners
we need to take into account those records grouped as "Missing partner".

TT47771

victoralmau committed:
[IMP] account_financial_report: Change Missing Partner as translatable

TT47771

eLBati committed:
[FIX] account_financial_report: show currency name instead of dictionary

Steps:

 - Accounting > Reporting > OCA accounting reports > Journal Ledger
 - Select Date range and activate Foreign Currency

Get "{'name': 'EUR'} " in currency field

Desired: get EUR

victoralmau committed:
[FIX] account_financial_report: Filter correctly by account_type in General ledger

In v15 it was filtered by the internal_type field (type of account.account.type) to use
"other" (most of the accounts had it set). Now in v16, there is the account_type field
exists and has to be compatible. The in or not in operator is set to "simplify" the
domain and avoid having to set almost all types.

norlinhenrik committed:
[IMP] account_financial_report: reversed variable only_one_unaffected_earnings_account

norlinhenrik committed:
[FIX] account_financial_report: report_action javascript

Commit:
[UPD] Update account_financial_report.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/

carolinafernandez-tecnativa committed:
[17.0][IMP] account_financial_report: Set custom intervales in aged report

mymage committed:
Translated using Weblate (Italian)

Currently translated at 98.4% (248 of 252 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/it/

tate11 committed:
Translated using Weblate (Turkish)

Currently translated at 76.9% (194 of 252 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/tr/

Commit:
[UPD] Update account_financial_report.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (267 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/es/

pilarvargas-tecnativa committed:
[IMP] account_financial_report: Change font size

Set a larger font style for the html report with a
relative font size to make it responsive. For this,
a new html_container is defined so that this does
not affect the other reports and the font style is
customised only for the report in html view.

TT48995

mymage committed:
Translated using Weblate (Italian)

Currently translated at 99.2% (265 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/it/

Saran440 committed:
[FIX] account_financial_report: fix error false due_date

victoralmau committed:
[IMP] account_financial_report: Change t-esc and t-raw to t-out from Trial Balance report

TT48969

OCA-git-bot committed:
[BOT] post-merge updates

Saran440 committed:
[FIX] account_financial_report: aged partner compute due date

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 99.2% (265 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/it/

OCA-git-bot committed:
[BOT] post-merge updates

jakobkrabbe committed:
Translated using Weblate (Swedish)

Currently translated at 100.0% (267 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/sv/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 99.2% (265 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 99.6% (266 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/it/

vvrossem committed:
Replace 'Camptocamp SA' with 'Camptocamp'

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (267 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/es/

alexis-via committed:
[FIX] account_financial_report: crash in XSLX report on partner balance

pedrocs-exo committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (267 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/pt/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (267 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/it/

victoralmau committed:
[IMP] account_financial_report: Add grouping by analytical account in Trial balance

TT48969

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (267 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/it/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (267 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/it/

carolinafernandez-tecnativa committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (267 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/es/

carolinafernandez-tecnativa committed:
[FIX] account_financial_report: Export to XLSX after report preview

Fix ReferenceError: _ is not defined when printing report to xlsx after preview

OCA-git-bot committed:
[BOT] post-merge updates

SilvioC2C committed:
[FIX] account_financial_report: fix ValueError

``analytic_distribution`` is a JSON field where keys are 'account.id' or 'account.id,account.id'
Eg: https://github.com/odoo/odoo/blob/8479b4e/addons/sale/models/account_move_line.py#L158
Previous implementation tried to convert to ``int`` the ``analytic_distribution`` dict keys, leading to a ``ValueError``.
This commit fixes this bug.

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (267 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/it/

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[FIX] account_financial_report: Display the values of Amount cur. + Cumul cur. values in general ledger only if necessary

The currency_id field of the lines is always set, so follow the same logic as
in report/templates/general_ledger and show the data only if the currency is set
in the account.

TT51204

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] account_financial_report: Add Amount cur. to General Ledger totals if the account has not set currency

Related to https://github.com/OCA/account-financial-reporting/pull/1235#issuecomment-2419269015

[FIX] account_financial_report: Avoid error in General ledger if the Show foreign currency checkbox is not checked.

Related to https://github.com/OCA/account-financial-reporting/pull/1236

TT51996

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (267 of 267 strings)

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/it/

alexis-via committed:
[FIX] account_financial_report: in XLSX report, display null ending balance (just like we display null initial balance)

alexis-via committed:
[FIX] account_financial_report: Open Items report now sorted by account code

alexis-via committed:
[IMP] account_financial_report: avoid crash if data structure changes in the future

florian-dacosta committed:
[FIX] account_financial_report : fix wizard model names sent to qweb reports for vat and aged balance reports

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] account_financial_report: Change t-esc and t-raw to t-out from General Ledger report

Commit:
[UPD] Update account_financial_report.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: account-financial-reporting-17.0/account-financial-reporting-17.0-account_financial_report
Translate-URL: https://translation.odoo-community.org/projects/account-financial-reporting-17-0/account-financial-reporting-17-0-account_financial_report/

chaule97 committed:
[IMP] account_financial_report: pre-commit auto fixes

chaule97 committed:
[MIG] account_financial_report: Migration to 18.0

natuan9 commented on account_financial_report/report/aged_partner_balance_xlsx.py:
Looks like you forgot to replace it with `self.env._` for the files in the report folder

sebalix commented:
@chaule97 `date_range` and `report_xlsx` deps have been merged into 18.0

OCA-git-bot committed:
[BOT] post-merge updates

yajo committed:
[FIX] account_financial_report: KeyError rendering report

Some reports were failing to render without this fix.

It seems there was a regression introduced in 0a2b2736cdc1a73d6248b91fa79a0f4c6aedc99b.

@moduon MT-8285
